src/net_processing.cpp

```diff
@@ -1880,6 +1883,98 @@ void static ProcessOrphanTx(CConnman* connman, std::set<uint256>& orphan_work_se
     }
 }
 
+/**
+ * Common validation logic for GETCFILTER, GETCFHEADER, GETCFCHECKPT request handling. If the
+ * request is valid and can be serviced, this returns the stop block index and the filter index
+ * as out parameters. May disconnect from the peer in the case of a bad request.
+ */
+static bool PrepareBlockFilterRequest(CNode* pfrom, const CChainParams& chain_params,
+                                      BlockFilterType filter_type, uint32_t start_height,
+                                      uint256& stop_hash, uint32_t max_height_diff,
+                                      const CBlockIndex*& stop_index,
+                                      BlockFilterIndex*& filter_index)
+{
+    bool supported_filter_type =
+        (filter_type == BlockFilterType::BASIC &&
+         (pfrom->GetLocalServices() & NODE_COMPACT_FILTERS));
```

Comment:
aaceafd

Is generic BIP 157 support can be checked at beginning of ProcessMessage like it's done for NODE_BLOOM ? It would avoid to busy ThreadMessageHandler for nothing?